### PR TITLE
Twin validator test

### PR DESCRIPTION
**Pull request by zjma**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4121)

### CODE REVIEW

File: testsuite/forge-cli/src/main.rs [hunk 1]
```diff
@@ -645,11 +646,43 @@ fn single_test_suite(test_name: &str) -> Result<ForgeConfig<'static>> {
             },
             false,
         ),
+        "twin_validator_test" => config
+            .with_network_tests(vec![&TwinValidatorTest])
+            .with_initial_validator_count(NonZeroUsize::new(20).unwrap())
+            .with_initial_fullnode_count(5)
+            .with_genesis_helm_config_fn(Arc::new(|helm_values| {
+                // Have single epoch change in land blocking
```
Comment (sitalkedia) [inline testsuite/forge-cli/src/main.rs:654]: This comment is not applicable.

**Review by sitalkedia — COMMENTED** [review]
Looks great overall % minor nits - thank you for working on this.

File: testsuite/testcases/src/twin_validator_test.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    .swarm()
+                    .validator_mut(main_id)
+                    .unwrap()
+                    .get_identity()
+                    .await
+                    .expect(format!("Error while getting identity for {main_id}").as_str());
+                ctx.swarm()
+                    .validator_mut(main_id)
+                    .unwrap()
+                    .stop()
```
[the remark this answers is not in the document]
Comment (zjma) [reply testsuite/testcases/src/twin_validator_test.rs:57]: @sitalkedia it was stopped [here](https://github.com/aptos-labs/aptos-core/blame/ee0624f948211bbebc641bfca5151b6c746a2685/testsuite/cluster-test/src/experiments/twin_validator.rs)

File: .github/workflows/twin-validator-test.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  schedule:
+    - cron: "0 */3 * * *"
+
+jobs:
+  run-twin-validator-test:
+    uses: ./.github/workflows/run-forge.yaml
+    secrets: inherit
+    with:
+      FORGE_NAMESPACE: forge-twin-validator
+      FORGE_CLUSTER_NAME: aptos-forge-big-1
```
Comment (sitalkedia) [inline .github/workflows/twin-validator-test.yaml:18]: Can you add `FORGE_RUNNER_DURATION_SECS` here? May be just run this test for 15 mins.

File: testsuite/forge-cli/src/main.rs [hunk 4]
```diff
@@ -649,6 +650,33 @@ fn single_test_suite(test_name: &str) -> Result<ForgeConfig<'static>> {
...
         ),
+        "twin_validator_test" => config
+            .with_network_tests(vec![&TwinValidatorTest])
+            .with_initial_validator_count(NonZeroUsize::new(20).unwrap())
+            .with_initial_fullnode_count(5)
+            .with_genesis_helm_config_fn(Arc::new(|helm_values| {
+                helm_values["chain"]["epoch_duration_secs"] = 300.into();
+            }))
+            .with_success_criteria(SuccessCriteria::new(
+                if duration.as_secs() > 1200 {
```
Comment (sitalkedia) [inline testsuite/forge-cli/src/main.rs:661]: I would just hardcode the duration in `FORGE_RUNNER_DURATION_SECS` and have a fixed threshold instead.

**Review by sitalkedia — APPROVED** [review]
LGTM

File: .github/workflows/twin-validator-test.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  issues: write
+  pull-requests: write
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 */3 * * *"
+
+jobs:
+  run-twin-validator-test:
```
Comment (rustielin) [inline .github/workflows/twin-validator-test.yaml:13]: nit: rename to something like `run-forge-twin-validator-test` so it's easier to grep forge continuous tests :)

**Review by rustielin — APPROVED** [review]
Awesome work!

**Pull request merged**